### DISCUSSION TITLE
How to use emotion v10 with next.js

### DISCUSSION THREAD

**Discussion by bcye** [vercel/next.js#15289] [Help] [answered]
I'm switching from Gatsby to Next, because of SSR and am a bit confused about how to use emotion.
I already have emotion set up and used throughout my project, so I want to keep using Emotion.

I first followed the "with-emotion" example, which didn't work for me. I then saw on Emotion's page that "This only applies if you’re using vanilla Emotion or a version of Emotion prior to v10. For v10 and above, SSR just works in Next.js." and thought no additional set up is required, so I removed emotion and emotion-server from my project, but the styles still didn't work.

How can I use emotion with next?

**Comment by juanbzz** [comment]
@bcye Emotion v10 should work out of the box with Next.js. You install it, import it, and start using it. It's been working for me like that.

Please share a minimal working example using codesandbox.io or a repository.

**Reply by bcye** [reply]
Does this apply to the css prop as well?

**Comment by bcye** [comment, marked as answer]
I was using the css prop, which seems to not be supported by default.

I fixed it by [customising the babel config](https://nextjs.org/docs/advanced-features/customizing-babel-config):
```json
{
  "presets": ["next/babel", "@emotion/babel-preset-css-prop"],
  "plugins": ["emotion"]
}
```

I think only the preset is required, but the plugin is recommended.

Here are the links on how to install them:
- https://emotion.sh/docs/@emotion/babel-preset-css-prop
- https://emotion.sh/docs/install#with-babel-plugin-emotion

**Reply by IRediTOTO** [reply]
Can you tell me how to extract all of that inline-css to a seperate css file ?

**Reply by bcye** [reply]
@IRediTOTO you could look through the sources in your browser, I guess there should be a generated file.

**Reply by IRediTOTO** [reply]
Hi thank you for replie me, I tried but not work. I don't want emotion run on client, I wish export all css in to a file, run css-in-js make lower point on google page speed... Images bellow show css genarate by JS, dont have any css file loaded But it still styled.

![image](https://user-images.githubusercontent.com/15695252/94578713-40336680-02a2-11eb-965c-9640df8e44ab.png)

![image](https://user-images.githubusercontent.com/15695252/94578419-e9c62800-02a1-11eb-9a30-f04126a35805.png)

**Reply by adamk33n3r** [reply]
> I was using the css prop, which seems to not be supported by default.
> 
> I fixed it by [customising the babel config](https://nextjs.org/docs/advanced-features/customizing-babel-config):
> 
> ```json
> {
>   "presets": ["next/babel", "@emotion/babel-preset-css-prop"],
>   "plugins": ["emotion"]
> }
> ```
> 
> I think only the preset is required, but the plugin is recommended.
> 
> Here are the links on how to install them:
> 
> * https://emotion.sh/docs/@emotion/babel-preset-css-prop
> * https://emotion.sh/docs/install#with-babel-plugin-emotion

When I try this, I get a `ReferenceError: React is not defined` when compiling.

**Comment by samuelhnrq** [comment]
With react 17 the JSX transform [has changed](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) so did the [babel plugin's](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx#react-automatic-runtime) the new config for setting up emotion is "importSource" of the automatic JSX runtime.

```json
{
  "presets": [
    [
      "next/babel",
      {
        "preset-react": {
          "importSource": "@emotion/react"
        }
      }
    ]
  ]
}
```

The config is only needed for not having to add the new `/** @jsxImportSource @emotion/react */` to the top of every file. It actually works right out of the box.

Sorry for necrobumping but this is top result for google search

**Reply by pooldeveloper** [reply]
it works perfectly!
